sources/packages/api/src/route-controllers/student-appeal/student-appeal.students.controller.ts

```diff
@@ -0,0 +1,119 @@
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, UserToken } from "../../auth/decorators";
+import { IUserToken } from "../../auth/userToken.interface";
+import {
+  ApiTags,
+  ApiCreatedResponse,
+  ApiNotFoundResponse,
+  ApiUnprocessableEntityResponse,
+  ApiBadRequestResponse,
+} from "@nestjs/swagger";
+import BaseController from "../BaseController";
+import { ClientTypeBaseRoute, ApiProcessError } from "../../types";
+import { INVALID_APPLICATION_NUMBER, INVALID_FORM_DATA } from "../../constants";
+import { CustomNamedError } from "../../utilities";
+
+@AllowAuthorizedParty(AuthorizedParties.student)
+@Controller("appeal")
+@ApiTags(`${ClientTypeBaseRoute.Student}-appeal`)
+export class StudentAppealStudentsController extends BaseController {
+  constructor(
+    private readonly studentAppealService: StudentAppealService,
+    private readonly applicationService: ApplicationService,
+    private readonly formService: FormService,
+  ) {
+    super();
+  }
+
+  /**
+   * Submit student appeal.
+   * @param applicationId application for which the appeal is submitted.
+   * @param payload student appeal with appeal requests.
+   * @param userToken
+   */
+  @ApiCreatedResponse({ description: "Student appeal created successfully." })
+  @ApiNotFoundResponse({
+    description:
+      "Application either not found or not eligible to request an appeal.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description: "There is an existing appeal for this student.",
+  })
+  @ApiBadRequestResponse({
+    description: "Not able to submit student appeal due to invalid request.",
+  })
+  @Post("application/:applicationId")
+  async submitStudentAppeal(
+    @Param("applicationId") applicationId: number,
+    @Body() payload: StudentAppealDTO,
+    @UserToken() userToken: IUserToken,
+  ): Promise<PrimaryIdentifierDTO> {
+    const application = this.applicationService.getApplicationToRequestAppeal(
+      userToken.userId,
+      undefined,
+      applicationId,
+    );
+    if (!application) {
+      throw new NotFoundException(
+        new ApiProcessError(
+          "Given application either does not exist or is not complete to request change.",
+          INVALID_APPLICATION_NUMBER,
+        ),
+      );
+    }
+    const existingStudentAppeal =
+      await this.studentAppealService.hasExistingAppeal(userToken.userId);
+    if (existingStudentAppeal) {
+      throw new UnprocessableEntityException(
+        "There is already a pending appeal for this student.",
+      );
+    }
+    try {
+      const dryRunPromise = payload.studentAppealRequests.map((appeal) =>
+        this.formService.dryRunSubmission(appeal.formName, appeal.formData),
+      );
+      const submissionResults = await Promise.all(dryRunPromise);
+      submissionResults.forEach((result) => {
+        if (!result.valid) {
+          throw new CustomNamedError(
```

Comment:
Using the `CustomNamedError` seems to be an overread in this case, IMO. We can use 2 try/catch and handles the 2 possible different exceptions.